saleor/graphql/order/types.py

```diff
@@ -1833,13 +1855,70 @@ def resolve_token(root: SyncWebhookControlContext[models.Order], info):
     @staticmethod
     @prevent_sync_event_circular_query
     def resolve_discounts(root: SyncWebhookControlContext[models.Order], info):
-        @allow_writer_in_context(info.context)
+        order = root.node
+
+        # Line-lvl voucher discounts are represented as OrderDiscount objects for order
+        # created from checkout.
+        def wrap_line_discounts_from_checkout(order_discounts):
+            if order.origin != OrderOrigin.CHECKOUT:
+                return order_discounts
+
+            def wrap_order_line(order_lines):
+                def wrap_order_line_discount(
+                    order_line_discounts: list[list[discount_models.OrderLineDiscount]],
+                ):
+                    artificial_order_discount = None
```

Comment:
Suggestion: Add some comments what's going on with the `artificial_order_discount` as it will be hard to find out in a month

Reply:
Added :)